### ISSUE TITLE
Initial Sync Very Slow

### ISSUE THREAD

**Issue by downystreet** [openethereum/openethereum#11753]
I have 1GB/s internet connection and I have been syncing the blockchain for the last 3 days and still not finished yet using ./openethereum --cache-size=3072. It started out fairly quickly at first but has since slowed to a crawl after reaching a certain block height. At one point I was even getting a message saying "slow state fetch for hash". I am connected to 38/50 peers. I am getting about 1 block or more per second and my internet is only using about 1mb/s or less download with other nodes running on the system. Is this normal?

**Comment by dvdplm** [comment]
Syncing is very disk IO dependent so your SSD is likely the bottleneck here (rather than your network bandwidth). There are also "bumps" in the chain where things slow down significantly for a (long) while.

> I'm on block #10111110

Unless you know you want to sync from block 0 you are likely better off trying a warp sync. Did you turn off warp sync explicitly? Syncing from scratch takes many days even on the best hardware.

> slow state fetch for hash 0x.

This means one of your peers is requesting data from your node and it took a while – 890ms – to serve it. It's not an error and no immediate cause for concern. If you see a LOT of them it might slow your syncing speed down a bit, but in general it's not a problem.

**Comment by downystreet** [comment]
It warp synced in the beginning and then started downloading the full blockchain which is what I wanted. I've restarted it and its warp syncing again but very slowly. The warp sync has been running for about an hour and is at 263/4092.

**Comment by downystreet** [comment]
The snapshot is stuck on 265/4092 now. It goes up the whole screen.

**Comment by downystreet** [comment]
What directory is the actual blockchain being stored in? I don't have a .ethereum file, I see the snapshot file in .local/share/openethereum/chains/ethereum/903........./ . In this folder there are two directories (snapshot , overlayrecent).

**Comment by sammy007** [comment]
Just give up. They are not capable of producing anything that just works. Every upgrade is a major pain in the ass. Sync was always a gambling with this client. Just use other services if you want to dump coins. If you need that junk for "development" just review your goals. There is literally no reasons to develop anything for/with ethereum, they are all scams, snake oils and just plain stupid.

**Comment by downystreet** [comment]
Its now been for days since my last post when I indicated that I was resyncing the blockchain with the --no-warp option. I am now on block 4,449,111. This is the slowest blockchain I have ever downloaded and I've downloaded several including bitcoin and monero. After looking over what was going on I have just today set the cache size using the --cach-size=22528 and it is now syncing a little faster. Where as previously I was using a cache size of around 1024 to 2048 and was syncing at a rate of 1 or 2 blocks per second I am now with 22GB of cache syncing at a rate of around 50-100 blocks/s. Seeing that I'm not even halfway through, I estimate that the full sync will take at least another 4 days with a 1GB/s internet connection 32GB RAM and SSD raid.

**Comment by dvdplm** [comment]
> overlayrecent/db is the blockchain?

You are correct. The folder name is a bit obscure but it matches one of a few different database operation modes.

**Comment by naiemk** [comment]
I also have the same problem here. I am running a warp sync. Running a rinkeby sync on google cloud and mainnet on digitaocean.
Snapshot gets loaded (no snapshot for rinkeby! is this expected?). But after snapshot the rest of the sync is snail speed.

Mainnet sync is average .4 blocks/second at which speed 80k blocks takes a lifetime to complete.

I have synced parity before on the same setup. I am wondering why I cannot sync neither testnet nor mainnet any more.
